File: apps/dot/frontend/src/lib/api/imageGen.ts
import { apiClient } from '@/lib/api/api-client'
import type { GetAccessToken } from '@/lib/api/api-client'

export type GeneratedImageDto = {
  mime_type: string
  data_base64: string
  width: number
  height: number
}

export type ImageGenerateResponse = {
  images: GeneratedImageDto[]
  prompt_used: string
  count: number
  usage: {
    cost_usd: number
    model: string
  }
}

export type ImageGenerateRequest = {
  prompt: string
  count?: number | null
  aspect_ratio?: string
  resolution?: string
}

export async function generateImages(
  body: ImageGenerateRequest,
  getAccessToken: GetAccessToken,
): Promise<ImageGenerateResponse> {
  return apiClient.post<ImageGenerateResponse>('/v1/images/generate', body, getAccessToken)
}

File: apps/dot/backend/app/schemas/image_gen.py
"""Esquemas Pydantic para POST /v1/images/generate."""
from __future__ import annotations

from pydantic import BaseModel, Field


class ImageGenerateRequest(BaseModel):
    prompt: str = Field(..., min_length=1, max_length=4000)
    count: int | None = Field(default=None, ge=1, le=8)
    aspect_ratio: str | None = "1:1"
    resolution: str | None = "1024x1024"


class GeneratedImageResponse(BaseModel):
    mime_type: str
    data_base64: str
    width: int
    height: int


class ImageGenerateUsageResponse(BaseModel):
    cost_usd: float
    model: str


class ImageGenerateResponse(BaseModel):
    images: list[GeneratedImageResponse]
    prompt_used: str
    count: int
    usage: ImageGenerateUsageResponse
